util/src/main/java/com/ticketmaster/servos/util/DialogUtils.java

```diff
@@ -0,0 +1,102 @@
+package com.ticketmaster.servos.util;
+
+import android.app.AlertDialog;
+import android.app.AlertDialog.Builder;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.DialogInterface.OnClickListener;
+import android.content.DialogInterface.OnShowListener;
+import android.text.Editable;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+
+public final class DialogUtils {
+
+    private DialogUtils() { }
+
+    /**
+     * Get a dialog that shows the provided title and an indeterminate spinner. You are responsible for showing and canceling it.
+     */
+    public static ProgressDialog getProgressDialog(Context context, int titleResId) {
+        ProgressDialog progressDialog = new ProgressDialog(context);
+        progressDialog.setTitle(titleResId);
+        progressDialog.setIndeterminate(true);
+        return progressDialog;
+    }
+
+    /**
+     * Get a dialog that shows the provided message and an "OK" button to dismiss.
+     */
+    public static AlertDialog getMessageDialog(Context context, String message) {
+        return new Builder(context)
+                .setMessage(message)
+                .setPositiveButton(android.R.string.ok, null)
+                .create();
+    }
+
+    /**
+     * Get a dialog that shows the provided message and has two buttons: OK and Cancel. The dialog will be dismissed if either button is pressed.
+     * @param onClickListener The listener to be called if OK is clicked.
+     */
+    public static AlertDialog getOkCancelDialog(final Context context,
+                                               String message,
+                                               OnClickListener onClickListener) {
+        return new Builder(context)
+                .setMessage(message)
+                .setPositiveButton(android.R.string.ok, onClickListener)
+                .setNegativeButton(android.R.string.cancel, null)
+                .create();
+    }
+
+    /**
+     * Get a dialog with a title and an EditText for the user to type in. Automatically opens the keyboard when shown.
+     * @param title The title of the window
+     * @param hint  The hint of the EditText if initialInputText is empty.
+     * @param initialInputText The text that will be set on the EditText initially.
+     * @param listener The listener that will be called when the user clicks "OK" and be given the text that they entered.
+     */
+    public static AlertDialog getInputDialog(final Context context,
+                                             String title,
```

Comment:
To go along with my last comment, imo it'd be nice to have a version of this method that takes String resources for title and hint, though initialInputText is probably fine without it if I'm understanding its use case correctly.